OpenRA.Game/Widgets/Widget.cs

```diff
@@ -524,8 +524,10 @@ public virtual void HideChild(Widget child)
 
 		public virtual void RemoveChildren()
 		{
-			while (Children.Count > 0)
-				RemoveChild(Children[Children.Count - 1]);
+			foreach (var child in Children.AsEnumerable().Reverse())
```

Comment:
perhaps add a comment
```
// PERF: Instead of removing children one by one, notify all and clear the list

Reply:
addressed in #20235

Reply:
Honestly that comment is pointless because clearing the collection should be a no-brainer. A slightly more reasonable comment would be why there is a `.Reverse()` in there.